internal/service/domain/tree/import_tree_test.go

```diff
@@ -0,0 +1,193 @@
+		imageRepo := storageMock.NewMockImageRepository(t)
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		expectedTree := getTestTrees()[0]
+
+		// Mock GetByCoordinates to simulate no tree found
+		treeRepo.EXPECT().GetByCoordinates(ctx, treeToImport.Latitude, treeToImport.Longitude).Return(nil, nil)
+
+		// Mock Create to confirm new tree is created
+		treeRepo.EXPECT().Create(ctx,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).Return(expectedTree, nil)
+
+		// When
+		err := svc.ImportTree(ctx, []*entities.TreeImport{treeToImport})
+
+		// Then
+		assert.NoError(t, err)
+	})
+
+	t.Run("should update an existing tree when matching coordinates and planting year are found", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		imageRepo := storageMock.NewMockImageRepository(t)
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		existingTree := getTestTrees()[0]
+		updatedTree := getTestTrees()[0]
+		updatedTree.Description = "Updated description"
+
+		// Mock GetByCoordinates to return an existing tree
+		treeRepo.EXPECT().GetByCoordinates(ctx, treeToImport.Latitude, treeToImport.Longitude).Return(existingTree, nil)
+
+		//Mock GetByID because it is called in update function
+		treeRepo.EXPECT().GetByID(ctx, existingTree.ID).Return(existingTree, nil)
+
+		// Mock Update to confirm tree is updated
+		treeRepo.EXPECT().Update(ctx, existingTree.ID,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).Return(updatedTree, nil)
+
+		// When
+		err := svc.ImportTree(ctx, []*entities.TreeImport{treeToImport})
+
+		// Then
+		assert.NoError(t, err)
+	})
+	t.Run("should delete and recreate a tree when planting years differ", func(t *testing.T) {
+		// Given
+		treeRepo := storageMock.NewMockTreeRepository(t)
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		imageRepo := storageMock.NewMockImageRepository(t)
+		treeClusterRepo := storageMock.NewMockTreeClusterRepository(t)
+		locator := service.NewMockGeoClusterLocator(t)
+
+		svc := NewTreeService(treeRepo, sensorRepo, imageRepo, treeClusterRepo, locator)
+
+		// Define existing tree and tree import data
+		existingTree := getTestTrees()[0]
+		treeToImport.PlantingYear = 2024
+
+		// Mock GetByCoordinates to return an existing tree
+		treeRepo.EXPECT().GetByCoordinates(ctx, treeToImport.Latitude, treeToImport.Longitude).Return(existingTree, nil)
+
+		//Mock GetByID because it is called in delete function
```

Comment:
Do we really need those comments? I think the code is self-explanatory enough 👍 We just need // given, / // when, // then